projects/scripts/adi_board.tcl

```diff
@@ -953,40 +982,42 @@ proc ad_cpu_interconnect {p_address p_name {p_intf_name {}}} {
     set use_smart_connect 0
   }
 
-  if {$sys_cpu_interconnect_index == 0} {
+  if {$i_str eq "M00"} {
 
     if {$use_smart_connect == 1} {
-      ad_ip_instance smartconnect axi_cpu_interconnect [ list \
+      ad_ip_instance smartconnect $interconnect_name [ list \
         NUM_MI 1 \
         NUM_SI 1 \
       ]
-      ad_connect sys_cpu_clk axi_cpu_interconnect/aclk
-      ad_connect sys_cpu_resetn axi_cpu_interconnect/aresetn
+      ad_connect sys_cpu_clk ${interconnect_name}/aclk
+      ad_connect sys_cpu_resetn ${interconnect_name}/aresetn
     } else {
-      ad_ip_instance axi_interconnect axi_cpu_interconnect
-      ad_connect sys_cpu_clk axi_cpu_interconnect/ACLK
-      ad_connect sys_cpu_clk axi_cpu_interconnect/S00_ACLK
-      ad_connect sys_cpu_resetn axi_cpu_interconnect/ARESETN
-      ad_connect sys_cpu_resetn axi_cpu_interconnect/S00_ARESETN
+      ad_ip_instance axi_interconnect $interconnect_name
+      ad_connect sys_cpu_clk ${interconnect_name}/ACLK
+      ad_connect sys_cpu_clk ${interconnect_name}/S00_ACLK
+      ad_connect sys_cpu_resetn ${interconnect_name}/ARESETN
+      ad_connect sys_cpu_resetn ${interconnect_name}/S00_ARESETN
     }
 
     if {$sys_zynq == 3} {
       ad_connect sys_cpu_clk sys_cips/m_axi_fpd_aclk
-      ad_connect axi_cpu_interconnect/S00_AXI sys_cips/M_AXI_FPD
-    }
-    if {$sys_zynq == 2} {
+      ad_connect ${interconnect_name}/S00_AXI sys_cips/M_AXI_FPD
+    } elseif {($p_sel eq "HPM0_FPD") && ($sys_zynq == 2)} {
```

Comment:
Perhaps nest 
```
} elseif {$sys_zynq == 2} {
  if {$p_sel eq "HPM0_FPD"} {
    (...)
  }
  (...)
}
```

Reply:
Thanks for reviewing! 

I went with a single line to stay consistent with `ad_mem_hpx_interconnect`.
I feel like it's also a bit more readable when we end up having other `sys_zynq` values.